_unittest_solvers/test_26_emit.py

```diff
@@ -1182,81 +1181,62 @@ def test_license_session(self, add_app):
         results = self.aedtapp.results
         revision = self.aedtapp.results.analyze()
 
-        receivers = revision.get_receiver_names()
-        modeRx = TxRxMode.RX
-        modeTx = TxRxMode.TX
-        modeEmi = ResultType.EMI
         self.aedtapp.set_units("Frequency", "GHz")
 
-        def get_best_rx_channel(results, receiver):
+        def do_run():
             domain = results.interaction_domain()
             rev = results.current_revision
-            # get interferers
-            interferers = rev.get_interferer_names()
-            best_emi = 300.0
-            best_band = None
-            best_freq = None
-
-            combinations_run = 0
-            
-            domain.set_receiver(receiver)
             interaction = rev.run(domain)
 
-            for interferer in interferers:
-                for tx_band in rev.get_band_names(interferer, modeTx):
-                    for tx_freq in rev.get_active_frequencies(interferer, tx_band, modeTx):
-                        domain.set_interferer(interferer, tx_band, tx_freq)
-                        rx_band = rev.get_band_names(receiver, modeRx)[0]
-                        for rx_freq in rev.get_active_frequencies(receiver, rx_band, modeRx):
-                            
-                            domain.set_receiver(receiver, rx_band, rx_freq)
-
-                            if best_band == None:
-                                best_band = rx_band
-                            if best_freq == None:
-                                best_freq = rx_freq
-
-                            try: 
-                                instance = interaction.get_instance(domain)
-                                if instance.has_valid_values():
-                                    emi = instance.get_value(modeEmi)
-                                    if emi < best_emi:
-                                        best_emi = emi
-                                        best_band = rx_band
-                                        best_freq = rx_freq
-                                else:
-                                    assert(f'No valid values found')
-                            except:
-                                assert("No results between " + interferer + ": " + tx_band + 
-                                       ": " + str(tx_freq) + " and " + receiver + ": " + 
-                                       rx_band + ": " + str(rx_freq))
-                            
-                            combinations_run += 1
-
-                            if combinations_run >= 20:
-                                return [best_band, best_freq]
-            return [best_band, best_freq]
+        number_of_runs = 5
+
+        # Do a run to ensure the license log exists
+        do_run()
+
+        # Find the license log for this process
+        appdata_local_path = os.getenv('LOCALAPPDATA')
```

Comment:
@bryankaylor - does this resolve to the same as your approach: `tempfile.gettempdir()`

If so, using tempfile might be better. It is more compact and has cross-platform support. We don't support Linux with EMIT yet, but  it's best to write things in a way that will minimize the rework when we do.

Reply:
@myoung301 - Thanks for the suggestion! I didn't know about that import/function, I've updated the PR using it. It leads to `C:\Users\<username>\AppData\Local\Temp`.